zhaquirks/tuya/ts0601_pool_sensor.py

```diff
@@ -0,0 +1,268 @@
+# This dynamic approach voids changes to EnchantedDevice
+EnchantedDeviceV2 = type(
+    "EnchantedDeviceV2",
+    (CustomDeviceV2, EnchantedDevice),
+    {"tuya_spell_data_query": True},
+)
+
+
+class TuyaPoolManufCluster(TuyaMCUCluster):
+    """Tuya Manufacturer cluster with automatic data point refresh logic."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self._update_timer_handle = None
+        self.check_interval = 60
+        self.next_refresh_interval = 0
+        self._loop = asyncio.get_running_loop()
+        self.handle_auto_update_check_change()
+
+    def handle_auto_update_cancel(self):
+        """Auto update timer cancel."""
+        if self._update_timer_handle:
+            self._update_timer_handle.cancel()
+            self._update_timer_handle = None
+
+    def handle_auto_update_setup_next_call(self, force_new_interval=False):
+        """Auto update schedule next update."""
+        tuya_cluster = self.endpoint.device.endpoints[1].in_clusters.get(
+            TuyaMCUCluster.cluster_id, None
+        )
+        if tuya_cluster and "auto_refresh_interval" in tuya_cluster.attributes_by_name:
+            interval = tuya_cluster.get("auto_refresh_interval", 0) * 60
+            # Check for a change to auto refresh number
+            if interval != self.next_refresh_interval:
+                self.handle_auto_update_cancel()
+                self.next_refresh_interval = interval
+                force_new_interval = True
+
+        if force_new_interval and self.next_refresh_interval > 0:
+            self.debug(
+                "using refresh interval of %d minutes", self.next_refresh_interval
+            )
+            self._update_timer_handle = self._loop.call_later(
+                self.next_refresh_interval, self.handle_auto_update_timer_wrapper
+            )
+
+    def handle_auto_update_check_change(self):
+        """Auto update schedule next interval check."""
+        self.handle_auto_update_setup_next_call()
+        self._loop.call_later(self.check_interval, self.handle_auto_update_check_change)
+
+    def handle_auto_update_timer_wrapper(self):
+        """Auto update handle refresh and schedule next update."""
+        self.create_catching_task(self.handle_auto_update())
+        self.handle_auto_update_setup_next_call(force_new_interval=True)
+
+    async def handle_auto_update(self):
+        """Auto update invoke data refresh command."""
+        tuya_cluster = self.endpoint.device.endpoints[1].in_clusters[
+            TuyaMCUCluster.cluster_id
+        ]
+        self.debug("sending refresh query command")
+        await tuya_cluster.command(TUYA_QUERY_DATA)
+
+
+(
+    TuyaQuirkBuilder("_TZE200_v1jqz5cy", "TS0601")
+    .device_class(EnchantedDeviceV2)
+    .replaces(TuyaPoolManufCluster)
+    .tuya_battery(dp_id=7, power_cfg=TuyaPowerConfigurationCluster, scale=1)
```

Comment:
We can use the newer style here, `.tuya_battery(dp_id=7, battery_type=BatterySize.Other, battery_qty=1, battery_voltage=30, scale=1)` Or whatever the real values are.